packages/ui/src/components/ChatMessage/AuthorRow.tsx

```diff
@@ -0,0 +1,45 @@
+import { utils } from '@tloncorp/shared';
+import * as client from '@tloncorp/shared/dist/client';
+import { SizableText, View, XStack } from 'tamagui';
+
+import { Avatar } from '../Avatar';
+import ShipName from '../ShipName';
+
+const RoleBadge = ({ role }: { role: string }) => {
+  return (
+    <View
+      borderRadius="$l"
+      backgroundColor="$secondaryBackground"
+      paddingHorizontal="$s"
+    >
+      <SizableText fontWeight="$s" color="$secondaryText" size="$s">
+        {role}
+      </SizableText>
+    </View>
+  );
+};
+
+export default function AuthorRow({
+  author,
+  sent,
+  roles,
+}: {
+  author: client.Contact;
+  sent: string;
+  roles?: string[];
+}) {
+  const date = new Date(sent);
+  const timeDisplay = utils.makePrettyDayAndDateAndTime(date);
+  const firstRole = roles?.[0];
+
+  return (
+    <XStack gap="$space.l">
```

Comment:
Space is redundant here, `gap` should automatically pull from space values.

```suggestion
    <XStack gap="$l">
```